internal/venice-common/src/main/java/com/linkedin/venice/writer/VeniceWriter.java

```diff
@@ -372,6 +363,28 @@ public VeniceWriter(
     }
   }
 
+  private PubSubProducerAdapter getProducerAdapter() {
+    return producerAdapterLazy.get();
+  }
+
+  private ThreadPoolExecutor getThreadPoolExecutor() {
+    return threadPoolExecutorLazy.get();
+  }
+
+  // Create a thread pool which can have max 2 threads.
+  // Except during VW start and close we expect it to have zero threads to avoid unnecessary resource usage.
```

Comment:
What does this mean? Do you mean when VW is really in use it will have at max 2 threads? Otherwise it is 0 thread because of lazy init?

Reply:
Yeah that's right. Unless needed there will be 0 threads. And these aren't created until we really need them.